carsus/io/cmfgen/base.py

```diff
@@ -0,0 +1,440 @@
+import numpy as np
+import pandas as pd
+import itertools
+import gzip
+import warnings
+from carsus.io.base import BaseParser
+
+
+# TODO: add `skiprows` parameter
+def find_row(fname, string1, string2='', how='both', num_row=False):
+    """Search strings inside plain text files and returns matching\
+    line or row number.
+
+    Parameters
+    ----------
+    fname : str
+        Path to plain text file.
+    string1 : str
+        String to search.
+    string2 : str
+        Extra string to search (default is '').
+    how : {'one', 'both', 'first'}
+        If 'both' search for string1 AND string2. If 'one' search for string1\
+            OR string2. If 'first' searches for 'string1' AND NOT string2\
+            (default is 'both').
+    num_row : bool
+        If true, returns row number instead (default is False).
+
+    Returns
+    -------
+    str or int
+        Returns matching line or row number.
+
+    """
+    with open(fname, encoding='ISO-8859-1') as f:
+        n = 0
+        for line in f:
+            n += 1
+
+            if how == 'one':
+                if string1 in line or string2 in line:
+                    break
+
+            if how == 'both':
+                if string1 in line and string2 in line:
+                    break
+
+            if how == 'first':
+                if string1 in line and string2 not in line:
+                    break
+
+        # In case there's no match
+        else:
+            n = None
+            line = None
+
+    if num_row is True:
+        return n
+
+    return line
+
+
+def parse_header(fname, keys, start=0, stop=50):
+    """Parse header from CMFGEN files.
```

Comment:
for all CMFGEN files?

Reply:
Sorry, I don't fully understand the question.